src/redux/books/books.js

```diff
@@ -1,47 +1,82 @@
-import { v4 as uuidv4 } from 'uuid';
+// import { v4 as uuidv4 } from 'uuid';
 
 // Actions
+const GET_BOOKS = 'bookstore-react/books/GET_BOOKS';
 const ADD = 'bookstore-react/books/ADD';
 const REMOVE = 'bookstore-react/books/REMOVE';
 
 // Reducer
-export default function booksReducer(
-  state = [
-    {
-      title: 'First Book',
-      author: 'First Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Second Book',
-      author: 'Second Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Third Book',
-      author: 'Third Author',
-      id: uuidv4(),
-    },
-  ],
-  action = {}
-) {
+export default function booksReducer(state = [], action = {}) {
   switch (action.type) {
+    case GET_BOOKS:
+      return action.payload;
     case ADD:
       return [...state, action.book];
     case REMOVE:
-      return state.filter((book) => book.id !== action.book.id);
+      return state.filter((book) => book.item_id !== action.book.item_id);
     default:
       return state;
   }
 }
 
 // Action Creators
-export const addBook = (book) => ({
-  type: ADD,
-  book,
-});
+export const getBooks = () => async (dispatch) => {
+  const booksFetch = await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/z4gzcApO94uwV3qBSQiu/books',
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    }
+  ).then((data) => data.json());
```

Comment:
- [ ] Great job so far 👏. From the project requirements, you are expected the modern way to manage project data using different functions offered by the `redux-toolkit` like [CreateAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) and CreateSlice. All information about getting started and some examples can be found on the [redux-toolkit](https://redux-toolkit.js.org/introduction/getting-started) official documentation. What I would suggest is **to refactor your codes to use the method stated in the project requirements** 😃 👍